# Hypothesis_Testing

In [8]:
#import the libraries
import pandas as pd
import scipy 
import numpy as np
from scipy import stats
from scipy.stats import ttest_ind
import random

## 1) Cutlets data

In [9]:
# Load dataset
cutlets = pd.read_csv("/Users/Admin/Documents/data science assignment/hypothesis testing/Cutlets.csv")

In [10]:
cutlets.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [11]:
cutlets.describe()

Unit A     Unit B
count  35.000000  35.000000
mean    7.019091   6.964297
std     0.288408   0.343401
min     6.437600   6.038000
25%     6.831500   6.753600
50%     6.943800   6.939900
75%     7.280550   7.195000
max     7.516900   7.545900

#### H0 = There is no significant difference in the diameter of the cutlet between two units.

#### H1 = There is a significant difference in the diameter of the cutlet between two units.

In [12]:
sample1 = random.choices(cutlets['Unit A'],k=10)
sample1

[7.1561,
 6.9438,
 6.9246,
 6.8394,
 6.8568,
 6.7794,
 6.8568,
 6.9952,
 7.4488,
 7.2783]

In [13]:
sample2 = random.choices(cutlets['Unit B'],k=10)
sample2

[6.7478,
 7.117999999999999,
 7.1522,
 7.422000000000001,
 6.8889,
 7.0992,
 7.1688,
 6.6606,
 6.811,
 6.73]

In [14]:
stats.ttest_ind(sample1,sample2)

Ttest_indResult(statistic=0.2695280716733338, pvalue=0.7905888359564833)

#### The p-value is 0.7905888359564833 it greater than 0.05. so H0 null hypothesis is accepted and h1 is rejected.Hence there is no significant difference in the diameter of the cutlet between two unit.

## 2) Labtat data

In [15]:
# Load dataset
labtat = pd.read_csv("/Users/Admin/Documents/data science assignment/hypothesis testing/labTAT.csv")
labtat

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0          185.35        165.53        176.70        166.13
1          170.49        185.91        198.45        160.79
2          192.77        194.92        201.23        185.18
3          177.33        183.00        199.61        176.42
4          193.41        169.57        204.63        152.60
..            ...           ...           ...           ...
115        178.49        170.66        193.80        172.68
116        176.08        183.98        215.25        177.64
117        202.48        174.54        203.99        170.27
118        182.40        197.18        194.52        150.87
119        182.09        215.17        221.49        162.21

[120 rows x 4 columns]

#### H0 = There is no difference in the average Turn Around Time (TAT) of reports of the difference laboratories.

#### H1 = There is a difference in the average Turn Around Time (TAT) of reports of the difference laboratories.

In [16]:
# oneway Anova
stats.f_oneway(labtat.iloc[:,0],labtat.iloc[:,1],labtat.iloc[:,2],labtat.iloc[:,3])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

#### Here p-value is 2.1156708949992414e-57 which is lesser than 0.05. so H0 is rejected and H1 is accepted. so there is a difference in the average Turn Around Time (TAT) of reports of the difference laboratories.

## 3) BuyerRatio data

In [17]:
# load dataset
BuyerRatio = pd.read_csv("/Users/Admin/Documents/data science assignment/hypothesis testing/BuyerRatio.csv")
BuyerRatio

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

#### H0 = male-female buyer rations are similar across regions.

#### Ha = male-female buyer rations are not similar across regions.

In [18]:
BuyerRatio1 = pd.DataFrame(BuyerRatio)

In [19]:
BuyerRatio1['all'] = BuyerRatio1.sum(axis=1)
BuyerRatio1

Observed Values  East  West  North  South   all
0           Males    50   142    131     70   393
1         Females   435  1523   1356    750  4064

In [20]:
count=pd.crosstab(BuyerRatio1['Observed Values'],BuyerRatio1['all'])

In [21]:
Chisquares_results=scipy.stats.chi2_contingency(count)
Chisquares_results

(0.0,
 1.0,
 1,
 array([[0.5, 0.5],
        [0.5, 0.5]]))

In [22]:
chisquare_results=[['','Test Statistic','p-value'],['Sample Data',Chisquares_results[0],Chisquares_results[1]]]
chisquare_results 

[['', 'Test Statistic', 'p-value'], ['Sample Data', 0.0, 1.0]]

#### Here p-value for BuyerRatio of male and female is 1.0 it is greater than 0.05. hence H0 Null hypothesis is accepted so male-female buyer rations are similar across regions.

## 4) CustomerOrderForm data

In [31]:
# Load dataset
customerorderfrom = pd.read_csv("/Users/Admin/Documents/data science assignment/hypothesis testing/Costomer+OrderForm.csv")

In [32]:
customerorderfrom

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

#### H0 = % of Defective of all countries are equal i.e %Phillippines= %Indonesia= %Malta= %India

#### H1 = At least one defective % is not equal

In [33]:
from sklearn.preprocessing import LabelEncoder

In [34]:
labelencoder_customerorderfrom1=LabelEncoder()
customerorderfrom=customerorderfrom.apply(LabelEncoder().fit_transform)

In [35]:
# Chisquare 
Chisquares_results=scipy.stats.chi2_contingency(customerorderfrom)
Chisquares_results

(112.74730231885849,
 1.0,
 897,
 array([[0.74793008, 0.73689052, 0.7424103 , 0.77276909],
        [0.74793008, 0.73689052, 0.7424103 , 0.77276909],
        [0.49862006, 0.49126035, 0.4949402 , 0.51517939],
        ...,
        [0.74793008, 0.73689052, 0.7424103 , 0.77276909],
        [0.99724011, 0.9825207 , 0.9898804 , 1.03035879],
        [0.49862006, 0.49126035, 0.4949402 , 0.51517939]]))

In [36]:
Chi_square=[['','Test Statistic','p-value'],['Sample Data',Chisquares_results[0],Chisquares_results[1]]]

In [37]:
Chi_square

[['', 'Test Statistic', 'p-value'], ['Sample Data', 112.74730231885849, 1.0]]

#### Here p-value for TeleCall of customerorderform is 1.0 it is greater than 0.05. Hence H0 Null hypothesis is accepted where % Defective of all counteries are equal.